### PR TITLE
Add methods inmutables, add withEncodeTags.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] add WithEncodeTags.
- [x] add methods() inmutables.
- [x] code coverage 100%.

### CODE REVIEW

**Review by Mister-42 — CHANGES_REQUESTED** [review]

File: src/Accordion.php [hunk 1]
```diff
@@ -27,7 +27,7 @@
  *
  * ```php
  * echo Accordion::widget()
- *     ->items([
+ *     ->witItems([
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Accordion.php:30]: done.

File: src/Accordion.php [hunk 2]
```diff
@@ -74,17 +75,152 @@ protected function run(): string
...
+    }
+
+    /**
+     * Allows you to enable or disable the encoding tags html.
+     *
+     * @param bool $value
+     *
+     * @return self
+     */
+    public function withencodeTags(bool $value): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Accordion.php]: done.

File: src/Accordion.php [hunk 3]
```diff
@@ -27,7 +26,7 @@
  *
  * ```php
  * echo Accordion::widget()
- *     ->items([
+ *     ->witItems([
```
Comment (samdark) [inline src/Accordion.php:29]: I think the name could be left as is.

Comment (terabytesoftw) [reply src/Accordion.php:29]: All the methods are immutable so they must have the prefix with, break BC but then we have the package ready.

**Comment by philippfrenzel** [comment]
seems to be the solution to #36 ?! Sorry for checking this pull request after posting the issue. Cheers Philipp

File: src/Accordion.php (hunk 3 above)
Comment (Mister-42) [reply src/Accordion.php:29]:
```suggestion
 *     ->withItems([
```

**Review by Mister-42 — CHANGES_REQUESTED** [review]

File: src/Modal.php [hunk 4]
```diff
@@ -279,168 +262,227 @@ public function bodyOptions(array $value): self
...
      *
-     * @param array $value
-     *
-     * @return $this
-     *
-     * {@see \Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self
      */
-    public function footerOptions(array $value): self
+    public function withEncodeTags(): self
```
Comment (samdark) [inline src/Modal.php:306]:
```suggestion
    public function encodeTags(): self
```

File: src/Nav.php [hunk 5]
```diff
@@ -125,17 +124,152 @@ protected function run(): string
...
+
+        return $new;
+    }
+
+    /**
+     * Whether to activate parent menu items when one of the corresponding child menu items is active.
+     *
+     * @return $this
+     */
+    public function withActivateParents(): self
```
Comment (samdark) [inline src/Nav.php:197]:
```suggestion
    public function activateParents(): self
```

File: src/Nav.php [hunk 6]
```diff
@@ -125,17 +124,152 @@ protected function run(): string
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
Comment (samdark) [inline src/Nav.php:257]:
```suggestion
    public function encodeTags(): self
```

File: src/NavBar.php [hunk 7]
```diff
@@ -329,71 +315,114 @@ public function screenReaderToggleText(string $value): self
...
-        return $this;
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
Comment (samdark) [inline src/NavBar.php:393]:
```suggestion
    public function encodeTags(): self
```

File: src/Progress.php [hunk 8]
```diff
@@ -193,26 +143,99 @@ public function label(?string $value): self
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
Comment (samdark) [inline src/Progress.php:176]:
```suggestion
    public function encodeTags(): self
```

File: src/Tabs.php [hunk 9]
```diff
@@ -383,12 +262,141 @@ public function renderTabContent(bool $value): self
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
Comment (samdark) [inline src/Tabs.php:280]:
```suggestion
    public function encodeTags(): self
```

File: src/Toast.php [hunk 10]
```diff
@@ -347,8 +205,126 @@ public function title(string $value): self
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
Comment (samdark) [inline src/Toast.php:219]:
```suggestion
    public function encodeTags(): self
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Accordion.php [hunk 11]
```diff
@@ -69,22 +69,152 @@ protected function run(): string
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Accordion.php]: done.

File: src/Alert.php [hunk 12]
```diff
@@ -137,38 +80,119 @@ public function body(?string $value): self
...
-        return $this;
+        return $new;
+    }
+
+    /**
+     * Allows you to enable encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Alert.php]: done.

File: src/Breadcrumbs.php [hunk 13]
```diff
@@ -235,8 +208,62 @@ public function options(array $value): self
...
-        return $this;
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Breadcrumbs.php]: done.

File: src/Button.php [hunk 14]
```diff
@@ -95,8 +99,22 @@ public function options(array $value): self
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Button.php]: done.

File: src/ButtonDropdown.php [hunk 15]
```diff
@@ -309,8 +247,98 @@ public function split(bool $value): self
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/ButtonDropdown.php]: done.

File: src/ButtonGroup.php [hunk 16]
```diff
@@ -112,38 +76,94 @@ protected function renderButtons(): string
...
-        return $this;
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/ButtonGroup.php]: done.

File: src/ButtonToolbar.php [hunk 17]
```diff
@@ -113,24 +88,69 @@ protected function renderButtonGroups(): string
...
-        return $this;
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/ButtonToolbar.php]: done.

File: src/Carousel.php [hunk 18]
```diff
@@ -56,21 +57,122 @@ protected function run(): string
...
+
+        return $new;
+    }
+
+    /**
+     * Animate slides with a fade transition instead of a slide. Defaults to `false`.
+     *
+     * @return $this
+     */
+    public function withCrossfade(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Carousel.php:95]: done.

File: src/Carousel.php [hunk 19]
```diff
@@ -56,21 +57,122 @@ protected function run(): string
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Carousel.php]: done.

File: src/Dropdown.php [hunk 20]
```diff
@@ -47,22 +49,112 @@ protected function run(): string
...
+
+        return $new;
+    }
+
+    /**
+     * Allows you to enable the encoding tags html.
+     *
+     * @return self
+     */
+    public function withEncodeTags(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Dropdown.php]: done.

File: src/Modal.php (hunk 4 above)
Comment (terabytesoftw) [reply src/Modal.php]: done.

File: src/Nav.php (hunk 5 above)
Comment (terabytesoftw) [reply src/Nav.php]: done.

File: src/Nav.php (hunk 6 above)
Comment (terabytesoftw) [reply src/Nav.php]: done.

File: src/NavBar.php (hunk 7 above)
Comment (terabytesoftw) [reply src/NavBar.php]: done.

File: src/Progress.php (hunk 8 above)
Comment (terabytesoftw) [reply src/Progress.php]: done.

File: src/Tabs.php (hunk 9 above)
Comment (terabytesoftw) [reply src/Tabs.php]: done.

File: src/Toast.php (hunk 10 above)
Comment (terabytesoftw) [reply src/Toast.php]: done.

File: src/Accordion.php [hunk 21]
```diff
@@ -69,22 +69,152 @@ protected function run(): string
...
     }
 
+    /**
+     * Whether to close other items if an item is opened. Defaults to `true` which causes an accordion effect.
+     *
+     * Set this to `false` to allow keeping multiple items open at once.
+     *
+     * @return $this
+     */
+    public function withoutAutoCloseItems(): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Accordion.php]: done.

File: README.md [hunk 22]
```diff
@@ -44,7 +44,7 @@ For example, the following
 single line of code in a view file would render a Bootstrap Progress plugin:
 
 ```php
-<?= Yiisoft\Yii\Bootstrap5\Progress::widget(['percent' => 60, 'label' => 'test']) ?>
+<?= Yiisoft\Yii\Bootstrap5\Progress::widget()->withPercent('60')->withLabel('test') ?>
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply README.md:47]: done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**